binders/kafka-binder/spring-cloud-stream-binder-kafka-core/src/main/java/org/springframework/cloud/stream/binder/kafka/properties/KafkaBinderConfigurationProperties.java

```diff
@@ -665,4 +672,35 @@ public KafkaProducerProperties getExtension() {
 
 	}
 
+	public static class Metrics {
+		/**
+		 * When set to true, the offset lag metric of each consumer topic is computed whenever the metric is accessed (default: true).
+		 * When set to false only the periodically calculated offset lag is used.
+		 * See {@link #scheduledOffsetLagComputationInterval} for more information.
+		 */
+		private boolean realtimeOffsetLagComputationEnabled = true;
+
+		/**
+		 * The interval in which the offset lag for each consumer topic is computed (default: 60 seconds).
+		 * This value is used whenever {@link #realtimeOffsetLagComputationEnabled} is disabled or the real time
+		 * computation is taking too long.
+		 */
+		private Duration scheduledOffsetLagComputationInterval = Duration.ofSeconds(60);
```

Comment:
How about simply `offsetLagMetricsInterval`?

Reply:
It's shorter and I don't think we can give a more self-explanatory name anyway :)